src/main/java/org/opensearch/flowframework/workflow/RegisterRemoteModelStep.java

```diff
@@ -31,89 +30,78 @@
 import static org.opensearch.flowframework.common.CommonValue.CONNECTOR_ID;
 import static org.opensearch.flowframework.common.CommonValue.DESCRIPTION_FIELD;
 import static org.opensearch.flowframework.common.CommonValue.FUNCTION_NAME;
-import static org.opensearch.flowframework.common.CommonValue.MODEL_CONFIG;
-import static org.opensearch.flowframework.common.CommonValue.MODEL_FORMAT;
 import static org.opensearch.flowframework.common.CommonValue.MODEL_GROUP_ID;
-import static org.opensearch.flowframework.common.CommonValue.MODEL_VERSION;
+import static org.opensearch.flowframework.common.CommonValue.MODEL_ID;
 import static org.opensearch.flowframework.common.CommonValue.NAME_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.REGISTER_MODEL_STATUS;
 
 /**
  * Step to register a remote model
  */
-public class RegisterModelStep implements WorkflowStep {
+public class RegisterRemoteModelStep implements WorkflowStep {
 
-    private static final Logger logger = LogManager.getLogger(RegisterModelStep.class);
+    private static final Logger logger = LogManager.getLogger(RegisterRemoteModelStep.class);
 
     private MachineLearningNodeClient mlClient;
 
-    static final String NAME = "register_model";
+    static final String NAME = "register_remote_model";
 
     /**
      * Instantiate this class
      * @param mlClient client to instantiate MLClient
      */
-    public RegisterModelStep(MachineLearningNodeClient mlClient) {
+    public RegisterRemoteModelStep(MachineLearningNodeClient mlClient) {
         this.mlClient = mlClient;
     }
 
     @Override
     public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
 
-        CompletableFuture<WorkflowData> registerModelFuture = new CompletableFuture<>();
+        CompletableFuture<WorkflowData> registerRemoteModelFuture = new CompletableFuture<>();
 
         ActionListener<MLRegisterModelResponse> actionListener = new ActionListener<>() {
             @Override
             public void onResponse(MLRegisterModelResponse mlRegisterModelResponse) {
-                logger.info("Model registration successful");
-                registerModelFuture.complete(
+                logger.info("Remote Model registration successful");
+                registerRemoteModelFuture.complete(
                     new WorkflowData(
                         Map.ofEntries(
-                            Map.entry("model_id", mlRegisterModelResponse.getModelId()),
-                            Map.entry("register_model_status", mlRegisterModelResponse.getStatus())
+                            Map.entry(MODEL_ID, mlRegisterModelResponse.getModelId()),
```

Comment:
Just to confirm. Remote model returns modelId instead of taskId, right?

Reply:
yes, its not an async call